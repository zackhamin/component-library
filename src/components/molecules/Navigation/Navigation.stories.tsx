import { CustomerSearch, Orders, Subscriptions } from 'Atoms/Icons';
import { Navigation, NavigationProps } from './Navigation';

export default {
	title: 'molecules/Navigation',
	component: Navigation,
};

export const NavigationStory = (args: NavigationProps) => (
	<Navigation {...args} />
);

NavigationStory.storyName = 'Navigation';
NavigationStory.args = {
	showIcons: true,
	showTitle: true,
	items: [
		{
			sortOrder: 0,
			icon: <CustomerSearch />,
			title: 'Customer Search',
			href: '/',
			isActive: true,
		},
		{
			sortOrder: 1,
			icon: <Orders />,
			title: 'Orders',
			href: '/',
		},
		{
			sortOrder: 2,
			icon: <Subscriptions />,
			title: 'Subscriptions',
			href: '/',
		},
	],
};
