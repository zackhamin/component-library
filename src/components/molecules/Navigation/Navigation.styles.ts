import { colors } from '../../../styles/colors';
import styled from 'styled-components';

const StyledNavigation = styled.ul`
	list-style: none;
	padding: 0;
	margin: 0;
	display: flex;
	flex-direction: column;
	gap: 0.8rem;
	width: 100%;

	a {
		font-family: 'BrandonGrotesque-Bold';
		font-size: 2rem;
		padding: 1.6rem;
		background: ${colors.emerald5};
		border-radius: 0.8rem;
		display: flex;
		align-items: center;
		gap: 1.6rem;
		transition: 0.2s;

		&.active,
		&:hover {
			background: #fcfdff; // no variable for this color
		}
	}
`;

export { StyledNavigation };
