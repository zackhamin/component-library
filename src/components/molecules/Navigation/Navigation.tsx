import classNames from 'classnames';
import { ReactNode } from 'react';

import { StyledNavigation } from './Navigation.styles';

export interface NavigationItem {
	sortOrder: number;
	icon: ReactNode;
	title: string;
	href: string;
	isActive?: boolean;
}

export interface NavigationProps {
	showIcons?: boolean;
	showTitle?: boolean;
	items: NavigationItem[];
}

const Navigation = ({
	showIcons = false,
	showTitle = true,
	items,
}: NavigationProps) => {
	return (
		<StyledNavigation>
			{items.map(({ sortOrder, icon, title, href, isActive }) => (
				<li key={sortOrder}>
					{/* TODO: Use polymorphic link component */}
					<a
						href={href}
						className={classNames({ active: isActive })}
						title={title}
					>
						{showIcons && icon}
						{showTitle && title}
					</a>
				</li>
			))}
		</StyledNavigation>
	);
};

export { Navigation };
