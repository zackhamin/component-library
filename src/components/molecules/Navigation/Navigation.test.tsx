import { Navigation } from '../Navigation';
import { render } from '@testing-library/react';

describe('Navigation', () => {
	const defaultProps = {
		showIcons: true,
		showTitle: true,
		items: [
			{
				sortOrder: 0,
				icon: 'https://via.placeholder.com/24x24',
				title: 'Customer Search',
				href: '/',
			},
			{
				sortOrder: 1,
				icon: 'https://via.placeholder.com/24x24',
				title: 'Order Search',
				href: '/',
			},
		],
	};

	it('renders correctly', () => {
		const { container } = render(<Navigation {...defaultProps} />);
		expect(container.firstChild).toMatchSnapshot();
	});
});
