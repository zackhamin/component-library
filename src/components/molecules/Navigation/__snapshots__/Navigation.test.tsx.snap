// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation renders correctly 1`] = `
.c0 {
  list-style: none;
  padding: 0;
  margin: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 0.8rem;
  width: 100%;
}

.c0 a {
  font-family: 'BrandonGrotesque-Bold';
  font-size: 2rem;
  padding: 1.6rem;
  background: #F2F9F8;
  border-radius: 0.8rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 1.6rem;
  -webkit-transition: 0.2s;
  transition: 0.2s;
}

.c0 a.active,
.c0 a:hover {
  background: #fcfdff;
}

<ul
  class="c0"
>
  <li>
    <a
      class=""
      href="/"
      title="Customer Search"
    >
      https://via.placeholder.com/24x24
      Customer Search
    </a>
  </li>
  <li>
    <a
      class=""
      href="/"
      title="Order Search"
    >
      https://via.placeholder.com/24x24
      Order Search
    </a>
  </li>
</ul>
`;
